resqpy/property/grid_property_collection.py

```diff
@@ -1044,3 +747,402 @@ def write_nexus_collection(self,
+    # todo: add kh and some other property kinds
+
+    return source_rv, source_ntg, source_poro, source_sat, source_perm
+
+
+def _extend_imported_coarsen_rock_volume(source_rv, other, box, collection, realization, uncache_other_arrays,
+                                         coarsening, copy_all_realizations):
+    # bulk rock volume
+    fine_rv_array = coarse_rv_array = None
+    if source_rv.number_of_parts() == 0:
+        log.debug('computing bulk rock volume from fine and coarse grid geometries')
+        source_rv_array = other.support.volume()
+        if box is None:
+            fine_rv_array = source_rv_array
+        else:
+            fine_rv_array = source_rv_array[box[0, 0]:box[1, 0] + 1, box[0, 1]:box[1, 1] + 1, box[0, 2]:box[1, 2] + 1]
+        coarse_rv_array = collection.support.volume()
+    else:
+        for (part, info) in source_rv.dict.items():
+            if not copy_all_realizations and info[0] != realization:
+                continue
+            fine_rv_array = _array_box(other, part, box = box, uncache_other_arrays = uncache_other_arrays)
+            coarse_rv_array = _coarsening_sum(coarsening, fine_rv_array)
+            _add_to_imported(collection, coarse_rv_array, 'coarsened from grid ' + str(other.support.uuid), info)
+
+    return fine_rv_array, coarse_rv_array
+
+
+def _extend_imported_coarsen_ntg(source_ntg, other, box, collection, realization, uncache_other_arrays, coarsening,
+                                 copy_all_realizations, fine_rv_array, coarse_rv_array):
+
+    # net to gross ratio
+    # note that coarsened ntg values may exceed one when reference bulk rock volumes are from grid geometries
+    fine_ntg_array = coarse_ntg_array = None
+    for (part, info) in source_ntg.dict.items():
+        if not copy_all_realizations and info[0] != realization:
+            continue
+        fine_ntg_array = _array_box(other, part, box = box, uncache_other_arrays = uncache_other_arrays)
+        coarse_ntg_array = _coarsening_weighted_mean(coarsening,
+                                                     fine_ntg_array,
+                                                     fine_rv_array,
+                                                     coarse_weight = coarse_rv_array,
+                                                     zero_weight_result = 0.0)
+        _add_to_imported(collection, coarse_ntg_array, 'coarsened from grid ' + str(other.support.uuid), info)
+
+    return fine_ntg_array, coarse_ntg_array
+
+
+def _extend_imported_nrv_arrays(fine_ntg_array, coarse_ntg_array, fine_rv_array, coarse_rv_array):
+    if fine_ntg_array is None:
+        fine_nrv_array = fine_rv_array
+        coarse_nrv_array = coarse_rv_array
+    else:
+        fine_nrv_array = fine_rv_array * fine_ntg_array
+        coarse_nrv_array = coarse_rv_array * coarse_ntg_array
+    return fine_nrv_array, coarse_nrv_array
```

Comment:
Do these arrays need an `_add_to_imported()` ?

Reply:
As discussed - arrays are generated in memory only for coarsening of other properties